sway/tree/arrange.c

```diff
@@ -284,9 +304,27 @@ void arrange_children_of(struct sway_container *parent) {
 		return;
 	}
 
+	if (config->smart_gaps && workspace->current_gaps != (workspace->children->length <= 1 ? 0 : 1)) {
+		if (workspace->current_gaps == 0) {
+			container_damage_whole(workspace);
```

Comment:
Why do we need that?

Reply:
If smart gaps are on and a second window is added to a workspace, damage gets left in the outer borders. This clears it.

Reply:
TBH I don't think this is the right fix. We should just damage before _and_ after arranging. We only damage after right now.

Reply:
The problem is that adding a container (by opening a terminal for example) to a workspace with a single container already in it *only arranges the new container*. If smart_gaps were on in this situation, the workspace would need to apply outer gaps.

Reply:
Keep in mind there's going to be some significant changes once the atomic layout updates PR is completed and merged. The calls to add damage in response to layout changes will all be removed and done in a single place when the layout is applied. For that reason you might not want to try too hard to get it perfect here.